src/test/scala/mesosphere/marathon/core/appinfo/impl/AppInfoBaseDataTest.scala

```diff
@@ -492,5 +494,33 @@ class AppInfoBaseDataTest extends UnitTest with GroupCreation {
 
       Then("no exception was thrown so status was successfully fetched")
     }
+
+    "requesting Pod lastTaskFailure when one exists" in {
+      import concurrent.ExecutionContext.Implicits.global
+      val f = new Fixture
+      Given("A pod definition")
+      val taskFailure = TaskFailureTestHelper.taskFailure
+      f.taskFailureRepository.get(pod.id) returns Future.successful(Some(taskFailure))
+      val instance1 = {
+        val instance = fakeInstance(pod)(f)
+        val task = instance.tasksMap.head._2
+        val failedTask = task.copy(taskId = Task.Id(taskFailure.taskId))
+        instance.copy(tasksMap = instance.tasksMap + (failedTask.taskId -> failedTask))
+      }
+      f.instanceTracker.instancesBySpec() returns Future.successful(InstanceTracker.InstancesBySpec.forInstances(instance1))
+
+      And("an instance in the repo")
+      f.groupManager.podVersion(any, any) returns Future.successful(Some(pod))
+      f.marathonSchedulerService.listRunningDeployments() returns Future.successful(Seq.empty)
+
+      When("Getting pod status with last task failures")
+      val podStatus = f.baseData.podStatus(pod).futureValue
+
+      Then("we get the failure in the app info")
+      podStatus.terminationHistory.head.instanceID shouldEqual instance1.instanceId.idString
+
+      And("the taskFailureRepository should have been called to retrieve the failure")
+      verify(f.taskFailureRepository, times(1)).get(pod.id)
```

Comment:
Why should this be verified? I think this tests an implementation detail and is not required. All we care about is the `podStatus` returned. How that is constructed we do not care.